### PR TITLE
[RW-5274][risk=no] Check the legality of input tos version before submission

### PR DESCRIPTION
**Pull request by YintongMa**
In this PR, we add a validator to check if the tos version is illegal in such cases:
1. null
2. not current term of service version

### CODE REVIEW

File: api/src/main/java/org/pmiops/workbench/db/dao/UserServiceImpl.java [hunk 1]
```diff
@@ -553,18 +553,29 @@ public void setDataUseAgreementNameOutOfDate(String newGivenName, String newFami
...
   }
 
-  @Override
-  @Transactional
-  public void submitTermsOfService(DbUser dbUser, Integer tosVersion) {
+  /**
+   * Validates a given tosVersion, by running all validation checks.
+   * 
+   * @param tosVersion
+   * @throws BadRequestException
```
Comment (yonghaoy) [inline api/src/main/java/org/pmiops/workbench/db/dao/UserServiceImpl.java:560]: personally I don't use @throws for unchecked exception. But that's up to you. If using, I think it usually followed by `if .....`
See go/java-practices/javadoc#throws

File: api/src/main/java/org/pmiops/workbench/db/dao/UserServiceImpl.java [hunk 2]
```diff
@@ -553,18 +553,29 @@ public void setDataUseAgreementNameOutOfDate(String newGivenName, String newFami
...
+   */
+  private void validateTermsOfServiceVersion(Integer tosVersion) {
```
Comment (yonghaoy) [inline api/src/main/java/org/pmiops/workbench/db/dao/UserServiceImpl.java:562]: nit: private method usually declared below public method. 
But this seems to be simple enough, I think you can just inline in the main function. Don't need to have separate method ..

**Review by yonghaoy — CHANGES_REQUESTED** [review]
Also revert the file changed by your IDE

File: api/src/test/java/org/pmiops/workbench/db/dao/UserServiceTest.java [hunk 3]
```diff
@@ -435,6 +436,20 @@ public void testSetBypassTimes() {
 
   @Test
   public void testSubmitTermsOfService() {
+    // Testing NULL input version
```
Comment (yonghaoy) [inline api/src/test/java/org/pmiops/workbench/db/dao/UserServiceTest.java:439]: suggest to put this in a new test method, something like `testSubmitTermsOfService_invalidVersion()`
We usually prefer each test method focus on one scenario, and keep it simple

File: api/src/main/java/org/pmiops/workbench/db/dao/UserServiceImpl.java (hunk 1 above)
Comment (YintongMa) [reply api/src/main/java/org/pmiops/workbench/db/dao/UserServiceImpl.java:560]: sorry, I don't get what do you mean by "if ....." ?

Comment (yonghaoy) [reply api/src/main/java/org/pmiops/workbench/db/dao/UserServiceImpl.java]: See the last sentence in go/java-practices/javadoc#throws
`When @throws SomeException is used, it is almost always followed by the word if and an explanation of the condition in which that exception is thrown`

**Review by yonghaoy — COMMENTED** [review]
Run `./gradlew spotlessApply ` to format code

**Review by yonghaoy — APPROVED** [review]

**Pull request merged**